### PR TITLE
Awesome Books Using ES6 Modules

### PR DESCRIPTION
**Pull request by rmiyoyo**
# Work Done
- I used ES6 modules to write modular JavaScript
- I implemented the time/date feature using Luxon JS
- Configured linters

### CODE REVIEW

**Review by Gambit142 — COMMENTED** [review]
Hi @rmiyoyo 👋, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has no linters checks running. Kindly setup linters properly 👍 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by rmiyoyo** [comment]
# Work Done
- I set up the linters files, the linter checks are now running.

File: index.html [hunk 1]
```diff
@@ -38,7 +38,7 @@ <h2 class="head">Contact Information</h2>
         </ul>
       </section>
       <section class="bookslistadd" class="content-section">
-        <h2 id="addbookinstruction">Add a New Book</h2>
+        <h2 id="addbookinstruction">Add New Book</h2>
```
Comment (Munsa1) [inline index.html:41]: - [ ] (Optional) Upon first load, your app is showing two sections in one view, it is showing the add book section and the book list section kindly fix this issue for improved user experience

Your app on first load
image

#### Note that the undefined book objects are book objects stored in localstorage by other apps. Fixing that bug would also improve the user experience of your app

image

File: modules/data.js [hunk 2]
```diff
@@ -22,7 +22,7 @@ export default class Books {
         }, 3000);
       }
       if (writer === '') {
```
Comment (Munsa1) [inline modules/data.js:24]: - [ ] The whole point of this milestone is to use `modules` as a way of learning how to do modular programming. Currently you only have 2 modules relevant to this project, `data.js` and `time.js`. Consider moving the `selectItems()` to a separate `.js` file. Also move the `removeItem = () => {}` function out of the `selectItems()` function into its separate `.js` file. Then you can import it back into the `selectItems()` function, and then import the `selectItems()` function back into `data.js` for a more modular approach.

image

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @rmiyoyo, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
Passing Linters :heavy_check_mark: 
Correct GitHub Flow :heavy_check_mark: 
Descriptive ReadMe File :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._
## Due to inssuficient files changed I will leave some comments here  :green_heart: 
### Important :red_circle:
- [ ] The `Luxon` package should added as a `.js` file to the project (in full or minified version). Kindly ensure you create a `.js` file for it and import it into your `time.js` component using, `import { DateTime } from "luxon";`
image

- [ ] In your ReadMe file kindly add a link which works to you `Live Demo` seection. The link should redirect to your deployed app. Optionally if you choose not to deploy, you can remove the `Live Demo` section from the ReadMe file and from its `table of contents`.
Your Live Demo Section
image

Where it Leads To
image


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by rmiyoyo** [comment]
# Work Done After First Review
- [x] Fixed the GitHub Pages deployment, the link now works.
- [x] Added more modules to the project.
- [x] Added luxon package for time.
- [x] Fixed functions to use the arrow syntax.
- [x] Ran linters and fixed linter errors.

File: modules/books.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    <hr class="divider"/>`;
+  });
+
+  document.querySelector('.all').innerHTML = markup;
+  removeItem();
+}
+
+export function addBook(bookheader, writer) {
+  const message = document.querySelector('.displayError');
+
```
Comment (abdulhamiid) [inline modules/books.js:39]: As requested by the previous reviewer and according to the [project requirements](https://github.com/microverseinc/curriculum-javascript/blob/main/books/books_with_es6.md), kindly refactor your functions to `arrow` functions. 💪

**Review by abdulhamiid — CHANGES_REQUESTED** [review]
# Status: Required Changes :recycle:
Hi @rmiyoyo 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, I really appreciate the effort you put in! but there are still some which haven't been addressed yet.

almost-there

## Highlights 🌟

- ✔️ Use ES6 modules to write modular JavaScript.
- ✔️ Use npm as a software package system.

## Required Changes ♻️
_Please kindly check the comments under the review._

## Optional suggestions 🤷‍♂️
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by rmiyoyo** [comment]
# Work Done After Second Review
- Refactored ```selectItems``` and ```removeItem``` functions to arrow functions.

**Review by Ghiftee — APPROVED** [review]
## STATUS: APPROVED ♻️♻️
Hi @rmiyoyo, 

To highlight 👍🏽 
✔️ Arrow functions are used

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀  
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**